precovery/precovery_db.py

```diff
@@ -509,3 +509,30 @@ def _check_frames(
 
                 n_frame += 1
             logger.info("checked %d frames", n_frame)
+
+    def find_observations_in_region(
+        self, ra: float, dec: float, obscode: str
+    ) -> Iterator[Observation]:
+        """Gets all the Observations within the same Healpixel as a
+        given RA, Dec for a particular observatory (specified as an obscode).
+
+        """
+        frames = self.frames.get_frames_for_ra_dec(ra, dec, obscode, self.config.nside)
+        for f in frames:
+            for o in self.frames.iterate_observations(f):
+                yield o
+
+    def find_observations_in_radius(
+        self, ra: float, dec: float, tolerance: float, obscode: str
```

Comment:
At some point, I would love to get pydantic style value validations to all these query inputs so that we and others get early warnings of nonsense coming in. Not really important for this PR but just wanted to mention it.